armi/nucDirectory/thermalScattering.py

```diff
@@ -42,228 +30,91 @@ class attributes on :py:class:`~armi.materials.material.Material` subclasses to
 
 The labels for these vary across evaluations (e.g. ENDF/B-VII, ENDF/B-VIII, etc.). We provide ENDF/B-III.0 and ACE
 labels. Other physics kernels will have to derive their own labels as appropriate in client code.
-
-Like NuclideBase and Element, we want to have only one ThermalScattering instance for each TSL, so we use a module-level
-directory called ``byNbAndCompound``. This improves efficiency and allows better cross-referencing when thousands of
-material instances would otherwise have identical instances of these.
-
-Thus, in practice, users should rarely instantiate these on their own.
 """
 
-from typing import Tuple, Union
-
-from armi.nucDirectory import elements
-from armi.nucDirectory import nuclideBases as nb
+from dataclasses import dataclass
 
+# strings that users might want to reference downstream
 BE_METAL = "Be-metal"
 BEO = "BeO"
-SIC = "SiC"
+CRYSTALLINE_GRAPHITE = "crystalline-graphite"
 D2O = "D2O"
+GRAPHITE_10P = "reactor-graphite-10P"
+GRAPHITE_30P = "reactor-graphite-30P"
 H2O = "H2O"
+SIC = "SiC"
 UN = "UN"
 UO2 = "UO2"
 ZRH = "ZrH"
-CRYSTALLINE_GRAPHITE = "crystalline-graphite"
-GRAPHITE_10P = "reactor-graphite-10P"
-GRAPHITE_30P = "reactor-graphite-30P"
-
-
-byNbAndCompound = {}
-
-
-class ThermalScattering:
-    """
-    Thermal Scattering data.
 
-    Parameters
+# thermal scattering label data
+BY_NAME_AND_COMPOUND = {
+    ("AL27", None): ("tsl-013_Al_027.endf", "al-27"),
+    ("BE", BE_METAL): (f"tsl-{BE_METAL}.endf", "be-met"),
+    ("BE", BEO): (BEO, "be-beo"),
+    ("C", CRYSTALLINE_GRAPHITE): (f"tsl-{CRYSTALLINE_GRAPHITE}.endf", "grph"),
+    ("C", GRAPHITE_10P): (f"tsl-{GRAPHITE_10P}.endf", "grph10"),
+    ("C", GRAPHITE_30P): (f"tsl-{GRAPHITE_30P}.endf", "grph30"),
+    ("C", SIC): ("tsl-CinSiC.endf", "c-sic"),
+    ("FE56", None): ("tsl-026_Fe_056.endf", "fe-56"),
+    ("H", H2O): ("tsl-HinH2O.endf", "h-h2o"),
+    ("H", ZRH): ("tsl-HinZrH.endf", "h-zrh"),
+    ("H2", D2O): (f"tsl-Din{D2O}.endf", "d-d2o"),
+    ("N", UN): ("tsl-NinUN.endf", "n-un"),
+    ("O", BEO): ("tsl-OinBeO.endf", "o-beo"),
+    ("O", D2O): (f"tsl-Oin{D2O}.endf", "o-d2o"),
+    ("O", UO2): ("tsl-OinUO2.endf", "o-uo2"),
+    ("SI", SIC): ("tsl-SIinSiC.endf", "si-sic"),
+    ("U", UN): ("tsl-UinUN.endf", "u-un"),
+    ("U", UO2): ("tsl-UinUO2.endf", "u-uo2"),
+    ("ZR", ZRH): ("tsl-ZRinZrH.endf", "zr-zrh"),
+}
+
+
+@dataclass
```

Comment:
I saw you had to remove `slots=True` for py3.8 support which is sad. I would recommend `frozen` as well to make these immutable
```suggestion
@dataclass(frozen=True)
```

Reply:
Actually, my commit message is a typo. We only have to support 3.9.

Still, that means no "slots".